### PR TITLE
Sending email with AWS SES 

### PR DESCRIPTION
**Pull request by jung1411**
Sends email to the user who creates account.

image

- Have not requested to moved out of sandbox on AWS yet as I was using my personal AWS account for testing. Right now, only verified emails can receive the email. If we request AWS to move out of sandbox, we do not need to verify emails in order to send them. Currently, the limit is 200 daily but it can also be increased when we request to move out of sandbox.

- html format will be changed but I just left it as is for now.

- the link will also be changed once we have other things set up.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,104 +1,136 @@
...
 .rts2_cache_umd/
 
 # Optional REPL history
+
 .node_repl_history
 
 # Output of 'npm pack'
-*.tgz
+
+\*.tgz
```
Comment (dhmoon91) [inline .gitignore:84]: ?

File: .gitignore [hunk 2]
```diff
@@ -1,104 +1,136 @@
...
 
 # Runtime data
+
 pids
-*.pid
-*.seed
-*.pid.lock
+_.pid
+_.seed
+\*.pid.lock
```
Comment (dhmoon91) [inline .gitignore:19]: why are these changed?

File: server/sendemail.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { SES } from "aws-sdk";
+import { SendEmailRequest } from "aws-sdk/clients/ses";
```
Comment (dhmoon91) [inline server/sendemail.ts:2]: oh they actually provide the type definition? cool

File: server/sendemail.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        },
+        Body: {
+          Html: {
+            Data: html,
+            Charset: charset,
+          },
+        },
+      },
+    };
+    ses.sendEmail(params).promise();
```
Comment (dhmoon91) [inline server/sendemail.ts:47]: why .promise? 
Make this function `async` and just put `await` here?

File: server/sendemail.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      <body>
+      <div style=font-family: Arial, Helvetica, sans-serif;">
+      <img src="kcsa_logo.png" alt="KCSA Logo" style="width:200px">
+      <h1>Welcome to KCSA!</h1>
+      <p>You have created an account with us at KCSA. To verify your email, please click the link below.</p>
+      <a href="http://kcsacanada.ca/">http://kcsacanada.ca/</a>
+      
+      </div>
+      </body>
+      </html>`;
```
Comment (dhmoon91) [inline server/sendemail.ts:25]: this is okay for now but could you look into how we can style out emails?

File: server/graphql/schema/mutations/createUser.ts [hunk 6]
```diff
@@ -31,6 +33,8 @@ export const createUser = {
   async resolve(parent: object, args: object) {
     const user = await User.create({ ...args, isAdmin: false });
 
+    sendEmail(user.email);
```
Comment (dhmoon91) [inline server/graphql/schema/mutations/createUser.ts:36]: this should be `await` when `sendEmail` becomes async func

File: server/sendemail.ts (hunk 5 above)
Comment (jung1411) [reply server/sendemail.ts:25]: okok

File: server/sendemail.ts (hunk 4 above)
Comment (jung1411) [reply server/sendemail.ts:47]: ok, i'll try that

File: server/graphql/schema/mutations/createUser.ts (hunk 6 above)
Comment (jung1411) [reply server/graphql/schema/mutations/createUser.ts:36]: okok

File: .gitignore (hunk 2 above)
Comment (jung1411) [reply .gitignore:19]: Oh I was creating separate aws_config.json so I added it in gitignore but then figured I could just use .env to store aws credentials so I erased  it but I guess it somehow changed the file itself.

File: .gitignore (hunk 1 above)
Comment (jung1411) [reply .gitignore:84]: https://github.com/koreancanadiansoccer/kcsaweb/pull/3#discussion_r706634410

File: .gitignore (hunk 2 above)
Comment (dhmoon91) [reply .gitignore:19]: pull this file from master

Comment (dhmoon91) [reply .gitignore:19]: `git checkout master .gitignore`

File: server/sendemail.ts (hunk 4 above)
Comment (dhmoon91) [reply server/sendemail.ts:47]: Also this email should be more dynamic.
We will have few different emails being sent out - but you can get that done later

File: .gitignore (hunk 2 above)
Comment (jung1411) [reply .gitignore:19]: okok

File: server/sendemail.ts (hunk 4 above)
Comment (jung1411) [reply server/sendemail.ts:47]: okok

Comment (jung1411) [reply server/sendemail.ts:47]: I tried async with await but it still can't send email for some reason.
I don't know why it only sends when I put promise() at the end.

Even tried putting the credentials from .env as parameters and still didn't work without promise.
```
const ses = new SES({
      accessKeyId: process.env.AWS_ACCESS_KEY_ID,
      secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
      region: REGION,
    });
```

Comment (dhmoon91) [reply server/sendemail.ts:47]: oh wait looks like `promise()` it self is callback function. 

Just leave it like this then.

**Review by dhmoon91 — APPROVED** [review]
